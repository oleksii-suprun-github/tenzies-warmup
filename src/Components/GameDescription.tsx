import React from 'react';
import Header from './UI/Header';
import RollDiceButton from './UI/RollDiceButton';

const GameDescription: React.FC = ({ startHandler }) => (
  <>
    <Header
      title="Tenzies"
      description="Roll until all dice are the same. Click each die to freeze it at its current value between
    rolls."
    />
    <RollDiceButton onClick={startHandler}>Start the game</RollDiceButton>
  </>
);
export default GameDescription;
