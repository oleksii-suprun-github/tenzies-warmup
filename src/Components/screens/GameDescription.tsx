import { MouseEventHandler } from 'react';
import { useTranslation } from 'react-i18next';
import Header from '../Header';
import RecordsTable from '../RecordsTable';
import DifficultySelector from '../DifficultySelector';
import RollDiceButton from '../RollDiceButton';

export interface GameDescriptionProps {
  startHandler: MouseEventHandler<HTMLButtonElement>;
  difficultyHandler: Function;
  records?: GameRecord[];
}

const GameDescription = ({ startHandler, difficultyHandler, records }: GameDescriptionProps) => {
  const { t, i18n } = useTranslation();

  const onClickHandler = () => {
    i18n.changeLanguage(i18n.language === 'en-GB' ? 'de-DE' : 'en-GB');
  };

  return (
    <>
      <button
        data-testid="language-switcher"
        onClick={onClickHandler}
        className="btn ml-auto flex w-auto justify-end bg-main-board text-3xl shadow-none xs-xl:absolute xs-xl:right-[10px] xs-xl:top-[10px] xl:static"
        aria-label={`${t('game.aria.langSwitcher')}`}
      >
        {i18n.language === 'en-GB' ? 'ðŸ‡©ðŸ‡ª' : 'ðŸ‡¬ðŸ‡§'}
      </button>
      <Header>
        <h1 className="mb-6 mt-5 text-4xl font-extrabold leading-none">ðŸŽ² Tenzies</h1>
        <p className="text-lg">{t('game.description')}</p>
      </Header>
      {!!records?.length && <RecordsTable data={records} />}
      <DifficultySelector difficultyHandler={difficultyHandler} />
      <RollDiceButton onClick={startHandler}>{t('game.start')}</RollDiceButton>
    </>
  );
};
export default GameDescription;
