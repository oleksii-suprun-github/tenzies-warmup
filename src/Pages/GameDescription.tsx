import React, { MouseEventHandler } from 'react';
import Header from '../components/Header';
import RecordsTable from '../components/RecordsTable';
import DifficultySelector from '../components/DifficultySelector';
import RollDiceButton from '../components/RollDiceButton';
import { Record } from 'types';

interface GameDescriptionProps {
  startHandler: MouseEventHandler<HTMLButtonElement>;
  difficultyHandler: Function;
  records: Record[];
}

const GameDescription: React.FC<GameDescriptionProps> = ({
  startHandler,
  difficultyHandler,
  records,
}) => (
  <>
    <Header>
      <h1>ðŸŽ² Tenzies</h1>
      <p>
        Roll until all dice are the same. Click each die to freeze it at its current value between
        rolls.
      </p>
    </Header>
    {!!records.length && <RecordsTable data={records} />}
    <DifficultySelector difficultyHandler={difficultyHandler} />
    <RollDiceButton onClick={startHandler}>Start the game</RollDiceButton>
  </>
);
export default GameDescription;
