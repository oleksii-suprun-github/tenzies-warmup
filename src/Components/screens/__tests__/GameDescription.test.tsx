import { render, screen } from '@testing-library/react';
import GameDescription, { GameDescriptionProps } from '../GameDescription';
import GameDescriptionMockProps from './__mocks__/GameDescriptionProps.json';

describe('GameDescription', () => {
  let defaultProps = {
    startHandler: () => {},
    difficultyHandler: () => {},
  };

  it('should render a component without records table', () => {
    const { container } = render(<GameDescription {...(defaultProps as GameDescriptionProps)} />);
    expect(container).toMatchSnapshot();
  });

  it('should render a component with records table', () => {
    let localProps = {
      ...defaultProps,
      ...GameDescriptionMockProps,
    };
    const { container } = render(<GameDescription {...(localProps as GameDescriptionProps)} />);

    const recordsList = screen.getByTestId('records-list');

    expect(container).toMatchSnapshot();
    expect(recordsList).toBeInTheDocument();
  });
});
