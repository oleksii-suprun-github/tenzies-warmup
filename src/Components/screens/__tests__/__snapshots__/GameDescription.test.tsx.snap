// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameDescription should render a component with records table 1`] = `
<div>
  <header
    class="text-center"
    id="header"
  >
    <h1
      class="mb-6 mt-5 text-4xl font-extrabold leading-none"
    >
      ðŸŽ² Tenzies
    </h1>
    <p
      class="text-lg"
    >
      Roll until all dice are the same. Click each die to freeze it at its current value between rolls.
    </p>
  </header>
  <section
    class="mt-[50px] flex w-full flex-col items-center justify-center"
    data-testid="records-list"
    id="records-list"
  >
    <h2
      class="mb-4 text-center text-xl font-extrabold"
    >
      Top 5 results:
    </h2>
    <div
      class="relative w-full overflow-x-auto"
      data-testid="records-table"
      id="records-table-responsive-wrapper"
    >
      <table
        class="table table-sm"
        id="records-table"
      >
        <thead
          class="bg-main-button text-main-die"
        >
          <tr
            class="border-separate text-center"
          >
            <th
              class="border-main-board-color border border-gray-400 text-base"
            >
              #
            </th>
            <th
              class="border-main-board-color border border-gray-400 text-base"
            >
              Date
            </th>
            <th
              class="border-main-board-color border border-gray-400 text-base"
            >
              Difficulty
            </th>
            <th
              class="border-main-board-color border border-gray-400 text-base"
            >
              Total clicks
            </th>
            <th
              class="border-main-board-color border border-gray-400 text-base"
            >
              Total time
            </th>
          </tr>
        </thead>
        <tbody
          class="text-center"
        >
          <tr
            class="record-element bg-white"
          >
            <td
              class="border-main-board-color border border-gray-400 p-1.5"
            >
              1
            </td>
            <td
              class="border-main-board-color border border-gray-400 p-1.5"
            >
              19th June at 15:44
            </td>
            <td
              class="border-main-board-color border border-gray-400 p-1.5"
            >
              Easy
            </td>
            <td
              class="border-main-board-color border border-gray-400 p-1.5"
            >
              9
            </td>
            <td
              class="border-main-board-color border border-gray-400 p-1.5"
            >
              19
              s
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
  <section
    class="mt=[35px] justify-center; flex w-full flex-col items-center"
    id="difficulty"
  >
    <h2
      class="mb-4 mt-10 text-center text-2xl font-extrabold"
    >
      Please choose the difficulty:
    </h2>
    <select
      class="select select-bordered"
      id="difficulty-selector"
      name="difficulty"
    >
      <option
        value="easy"
      >
        Easy
      </option>
      <option
        selected=""
        value="normal"
      >
        Normal
      </option>
      <option
        value="hard"
      >
        Hard
      </option>
    </select>
  </section>
  <button
    class="btn btn-primary mx-auto my-[25px] mt-[50px] w-fit min-w-32 bg-main-button text-2xl text-main-die"
    data-testid="roll-dice-button"
  >
    Start the game
  </button>
</div>
`;

exports[`GameDescription should render a component without records table 1`] = `
<div>
  <header
    class="text-center"
    id="header"
  >
    <h1
      class="mb-6 mt-5 text-4xl font-extrabold leading-none"
    >
      ðŸŽ² Tenzies
    </h1>
    <p
      class="text-lg"
    >
      Roll until all dice are the same. Click each die to freeze it at its current value between rolls.
    </p>
  </header>
  <section
    class="mt=[35px] justify-center; flex w-full flex-col items-center"
    id="difficulty"
  >
    <h2
      class="mb-4 mt-10 text-center text-2xl font-extrabold"
    >
      Please choose the difficulty:
    </h2>
    <select
      class="select select-bordered"
      id="difficulty-selector"
      name="difficulty"
    >
      <option
        value="easy"
      >
        Easy
      </option>
      <option
        selected=""
        value="normal"
      >
        Normal
      </option>
      <option
        value="hard"
      >
        Hard
      </option>
    </select>
  </section>
  <button
    class="btn btn-primary mx-auto my-[25px] mt-[50px] w-fit min-w-32 bg-main-button text-2xl text-main-die"
    data-testid="roll-dice-button"
  >
    Start the game
  </button>
</div>
`;
